src/ui/components/Header.js

```diff
@@ -42,13 +43,20 @@ class Header extends React.Component {
   }
 
   render() {
-    const { loading } = this.props;
+    const { loading, uploading } = this.props;
     const isLoaded = loading == 100;
+    const upload = uploading?.totalMB
+      ? `${(uploading?.uploadedMB / uploading?.totalMB) * 100}%`
+      : `${uploading?.uploadedMB}MB`;
+
     return (
       <>
         <div id="header">
           <div className="logo"></div>
+          {/* NOTE: Should we use the app state to set this status rather than relying on `element.innerText`? */}
           <div id="status"></div>
+          {/* NOTE: This needs a way to determine when the upload is completed: specifically when there is no `totalMB` */}
```

Comment:
when would we not have a total upload?

Perhaps this starts as uploaded and only changes if we learn about an uploading status?

Reply:
In `main.js` in the `onUploadedData` function `lengthMB` can be undefined. I'm not the circumstances under which that could happen.

In that case it not possible to know when the download is completed from the information provided. The has two downstream effects, 

1. Knowing when to remove the progress indicator based on upload completion. 
2. The antD progress indicator requires a percentage to now how much of the circle to colour in. If there is no `lengthMB` then we would need to use a spinner or some other component. I think the material-ui option could work but that requires adding their theme to the app and comes with a lot of bloat for a single component.